store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/getelements/handler/GetDataframeOfElementsHandler.java

```diff
@@ -32,42 +37,81 @@
+public class GetDataframeOfElementsHandler implements OutputOperationHandler<GetDataFrameOfElements, Dataset<Row>>, Serializable {
     private static final Logger LOGGER = LoggerFactory.getLogger(GetDataframeOfElementsHandler.class);
+    private static final long serialVersionUID = 6355341028414862100L;
 
     @Override
     public Dataset<Row> doOperation(final GetDataFrameOfElements operation,
                                     final Context context,
                                     final Store store) throws OperationException {
-        final User user = context.getUser();
+        final User user;
         final SparkSession spark;
-        if (user instanceof SparkUser) {
+        final Authorisations auths;
+        final String visibility;
+
+        if (context.getUser() instanceof SparkUser) {
+            user = context.getUser();
             spark = ((SparkUser) user).getSparkSession();
         } else {
             throw new OperationException("This operation requires the user to be of type SparkUser.");
         }
-        return doOperation(operation, (ParquetStore) store, spark);
 
+        if (user != null && user.getDataAuths() != null) {
+            auths = new Authorisations(user.getDataAuths().toArray(new String[user.getDataAuths().size()]));
+        } else {
+            auths = new Authorisations();
+        }
+
+        if (store.getSchema().getVisibilityProperty() != null) {
+            visibility = store.getSchema().getVisibilityProperty();
+        } else {
+            visibility = new String();
+        }
+
+        return doOperation(operation, (ParquetStore) store, spark, auths, visibility);
     }
 
     private Dataset<Row> doOperation(final GetDataFrameOfElements operation,
                                      final ParquetStore store,
-                                     final SparkSession spark) throws OperationException {
+                                     final SparkSession spark,
+                                     final Authorisations auths,
+                                     final String visibility) throws OperationException {
+        final FilterFunction<Row> filter;
         if (operation.getView().equals(store.getSchemaUtils().getEmptyView())) {
             LOGGER.debug("Retrieving elements as a dataframe");
             final String rootDir = store.getDataDir() + "/" + store.getGraphIndex().getSnapshotTimestamp() + "/";
+
+            if (!visibility.isEmpty()) {
+                filter = e -> isVisible(e, visibility, auths);
+            } else {
+                filter = e -> true;
+            }
             final Dataset<Row> dataset = spark
                     .read()
                     .option("mergeSchema", true)
-                    .parquet(rootDir + ParquetStoreConstants.GRAPH);
+                    .parquet(rootDir + ParquetStoreConstants.GRAPH)
+                    .filter(filter);
```

Comment:
The filter step should only be applied when required, I expect Spark would optimise it out anyway but we shouldn't require it to.